samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/DeliveryBean.java

```diff
@@ -4230,4 +4237,14 @@ public String getMinReqScale() {
       return ServerConfigurationService.getString("samigo.ajaxTimerMinReqScale","5000");
     }
 
+    public void calculateMinutesAndSecondsLeft() {
+        String ms = getAutoSaveRepeatMilliseconds();
+        int milliseconds = Integer.parseInt(ms);
```

Comment:
I don't know if the result of `getAutoSaveRepeatMilliseconds()` is gauranteed to return a String representation of an Integer, but should we protect against the possibility of a `NumberFormatException` being thrown here?

Reply:
Yes, it does: https://github.com/sakaiproject/sakai/blob/98b51d4d5936842ef3651678527a7f419c8cfcd0/samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/DeliveryBean.java#L3984